packages/calypso-e2e/src/jest-playwright-config/environment.ts

```diff
@@ -0,0 +1,210 @@
+		);
+
+		const logFilePath = path.join( this.testArtifactsPath, `${ this.testFilename }.log` );
+
+		// Start the browser.
+		const browser = await chromium.launch( {
+			...config.launchOptions,
+			logger: {
+				log: async ( name: string, severity: string, message: string ) => {
+					await fs.appendFile(
+						logFilePath,
+						`${ new Date().toISOString() } ${ process.pid } ${ name } ${ severity }: ${ message }\n`
+					);
+				},
+				isEnabled: ( name ) => name === 'api',
+			},
+		} );
+
+		// Proxy our E2E browser to bind our custom default context options.
+		// We need to it this way because we're not using the first-party Playwright
+		// and we can't use the playwright.config.js file.
+		// See https://playwright.dev/docs/api/class-testoptions.
+		const wpBrowser = new Proxy( browser, {
+			get: function ( target, prop: keyof Browser ) {
+				const orig = target[ prop ];
+
+				if ( typeof orig !== 'function' ) {
+					return orig;
+				}
+
+				if ( prop === 'newContext' ) {
+					return async function ( options: BrowserContextOptions ): Promise< BrowserContext > {
+						const context = await target.newContext( {
+							...options,
+							...config.contextOptions,
+							recordVideo: { dir: os.tmpdir() },
+						} );
+
+						await context.tracing.start( {
+							screenshots: true,
+							snapshots: true,
+						} );
+
+						return context;
+					};
+				}
+
+				if ( prop === 'newPage' ) {
+					return async function ( options: BrowserContextOptions ): Promise< Page > {
+						const page = await target.newPage( {
+							...options,
+							...config.contextOptions,
+							recordVideo: { dir: os.tmpdir() },
+						} );
+
+						const context = page.context();
+
+						await context.tracing.start( {
+							screenshots: true,
+							snapshots: true,
+						} );
+
+						return page;
+					};
+				}
+
+				return orig.bind( target );
+			},
+		} );
+
+		// Expose browser globally.
+		this.global.browser = wpBrowser;
+	}
+
+	/**
+	 * Handle events emitted by jest-circus.
+	 */
+	async handleTestEvent( event: Circus.Event ) {
+		switch ( event.name ) {
+			case 'test_start': {
+				// If a test has failed, skip rest of the steps.
+				if ( this.failure?.type === 'test' ) {
+					event.test.mode = 'skip';
+				}
+				// If a hook has failed, mark all subsequent test steps as failed.
+				// Handling is different compared to test steps because Jest treats
+				// failed hooks differently from tests.
+				if ( this.failure?.type === 'hook' ) {
+					// event.test.mode = 'fail';
+					// This is now commented out as it throws the "Type '"fail"' is not
+					// assignable to type 'BlockMode'" TS error. The accepted options are
+					// "skip", "todo", and "only".
+				}
+				break;
+			}
+			case 'hook_failure': {
+				this.failure = { type: 'hook', name: event.hook.type };
+				break;
+			}
+			case 'test_fn_failure': {
+				this.failure = { type: 'test', name: event.test.name };
+				break;
+			}
```

Comment:
Right now, because we are only concerned with skipping tests, there are two edge cases that can account for strange behaviot:

1. If a hook fails in a given block, we still try to run later describe blocks in the test. We probably don't want to do that, as those aren't going to be valid if none of the prior tests ran.

2. If a test fails, we will skip all future tests in future describe blocks, but we still run the _hooks_ in those future describe blocks. This can result in cases where we open new pages (see the liking post spec), but do nothing with them. This then muddies the artifacts, as you have videos and screenshots of just blank white screens of those secondary pages that were opened and immediately abandoned.

We can fix both of these by adding a check for failures at the start of each describe block:

```typescript
	case 'run_describe_start': {
		if ( this.failure ) {
			event.describeBlock.mode = 'skip';
		}
		break;
	}
```

Reply:
@dpasque I added this and triggered an intentional failure in the hook of a spec file, but the behavior observed does not change:

```
    ✕ Go to new post page
    ✕ Enter post title
    ✕ Image block: upload image file
    ✕ Image block: upload image file with reserved URL characters
    ✕ Audio block: upload audio file
    ✕ File block: upload audio file
    ✕ Publish and visit post
    ✕ Confirm Image block is visible in published post
    ✕ Confirm Image block is visible in published post (reserved name)
    ✕ Confirm Audio block is visible in published post
    ✕ Confirm File block is visible in published post
```

To be clear, with this change applied, the expectation is that all steps are marked `Skipped` instead, correct?

Reply:
I poked around a bit more, and I think I understand what you mean now - I was misunderstanding your comment fundamentally!

With the suggested changes applied, this is the result of `Likes: Post`:

```
    As the posting user
      ✕ Go to the new post page
      ✕ Enter post title
      ✕ Enter post text
      ✕ Publish and visit post
      ✕ Like post
      ✕ Unlike post
    As the liking user
      ○ skipped Go to the published post page
      ○ skipped Login via popup to like the post
```

Reply:
Yes! What you have there in the last comment is what I was thinking!